src/material/input/input.ts

```diff
@@ -318,6 +326,10 @@ export class MatInput
         this.stateChanges.next();
       });
     }
+
+    this.ngControl?.control?.events.pipe(takeUntilDestroyed(this._destroyRef)).subscribe(() => {
```

Comment:
This seems a bit iffy to me, what's the context behind it? I would assume this to be taken care of by the forms module.

Reply:
The `[formControl]` directive should be doing this since it updates the touched, valid, etc classes. This is a workaround until forms is fixed, otherwise using `<input matInput [formControl]="...">` in zoneless causes changed after checked errors

Reply:
Also this was supposed to be `markForCheck`, it autocompleted to `detectChanges` which I've fixed

Reply:
Should we leave a TODO to clean it up/provide context?

Reply:
Yeah, I added that as well